libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -87,8 +87,7 @@
 #endif // _CCCL_HAS_BUILTIN(__array_extent)
 
 #if _CCCL_HAS_BUILTIN(__builtin_assume_aligned) || (defined(_CCCL_COMPILER_MSVC) && _CCCL_MSVC_VERSION >= 1923) \
-  || defined(_CCCL_COMPILER_GCC) || defined(_CCCL_COMPILER_NVHPC) || defined(_CCCL_COMPILER_CLANG)              \
-  || defined(_CCCL_CUDA_COMPILER_NVCC)
```

Comment:
That is not the right fix, the builtin detection works fine, I did verify that it works will all host compilers including nvc++